app/component/SearchMainContainer.js

```diff
@@ -13,44 +14,55 @@ import SearchModalLarge from './SearchModalLarge';
 import Icon from './Icon';
 import { getAllEndpointLayers } from '../util/searchUtils';
 
+
 class SearchMainContainer extends React.Component {
   static contextTypes = {
     executeAction: React.PropTypes.func.isRequired,
     getStore: React.PropTypes.func.isRequired,
     router: React.PropTypes.object.isRequired,
+    location: React.PropTypes.object.isRequired,
     intl: intlShape.isRequired,
     breakpoint: React.PropTypes.string.isRequired,
   };
 
   static propTypes = {
     className: React.PropTypes.string,
+    searchModalIsOpen: React.PropTypes.bool.isRequired,
+    selectedTab: React.PropTypes.string.isRequired,
   }
 
-  state = {
-    selectedTab: 'destination',
-    modalIsOpen: false,
-  };
-
-  componentDidMount() {
-    this.context.getStore('SearchStore').addChangeListener(this.onSearchChange);
-  }
-
-  componentWillUnmount() {
-    this.context.getStore('SearchStore').removeChangeListener(this.onSearchChange);
+  componentDidUpdate() {
+    if (this.props.searchModalIsOpen) {
+      setTimeout(
```

Comment:
Did not actually test, the same trick was just used before. I will retest if it works fine without settimeout.